### PR TITLE
Updated for Wagtail 5.0

### PR DESCRIPTION
**Pull request by katdom13**
Release notes: https://docs.wagtail.org/en/stable/releases/5.0.html

- Code cleanup
- Updated CHANGELOG

`generic_chooser/templates/generic_chooser/_create_tab.html`
- Updated longrunning button accordingly: https://docs.wagtail.org/en/stable/releases/5.0.html#progress-button-button-longrunning-now-relies-on-data-attributes

All tests run OK

### CODE REVIEW

File: generic_chooser/widgets.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def get_create_item_url(self):
             return None
         else:
             return reverse(self.create_item_url_name)
-    
+
```
Comment (katdom13) [inline generic_chooser/widgets.py:63]: Removed whitespace

File: generic_chooser/templates/generic_chooser/_create_tab.html [hunk 2]
```diff
@@ -12,7 +14,15 @@
...
-                <button type="submit" class="button button-longrunning" data-clicked-text="{{ create_form_submitted_label }}"><span class="icon icon-spinner"></span><em>{{ create_form_submit_label }}</em></button>
+                {% wagtail_version as current_version %}
+                {% if current_version >= "5.0.0" %}
+                    <button type="submit" class="button button-longrunning" data-controller="w-progress" data-action="w-progress#activate" data-w-progress-active-value="{{ create_form_submitted_label }}">
+                        <span class="icon icon-spinner"></span>
+                        <em data-w-progress-target="label">{{ create_form_submit_label }}</em>
+                    </button>
+                {% else %}
+                    <button type="submit" class="button button-longrunning" data-clicked-text="{{ create_form_submitted_label }}"><span class="icon icon-spinner"></span><em>{{ create_form_submit_label }}</em></button>
+                {% endif %}
```
Comment (katdom13) [inline generic_chooser/templates/generic_chooser/_create_tab.html:25]: According to: https://docs.wagtail.org/en/stable/releases/5.0.html#progress-button-button-longrunning-now-relies-on-data-attributes

File: setup.py [hunk 3]
```diff
@@ -35,9 +35,13 @@
         'Framework :: Django',
         'Framework :: Django 3',
         'Framework :: Django 3.2',
-        'Framework :: Django 4.0',
```
Comment (katdom13) [inline setup.py:38]: Django 4.0 reaching EOLife

**Review by nickmoreton — APPROVED** [review]
Looks good to me.

**Review by gasman — APPROVED** [review]
Thanks @katdom13!

**Pull request merged**